src/main/kotlin/com/exactpro/th2/sim/template/rule/KotlinFIXRule.kt

```diff
@@ -29,35 +29,36 @@ import com.exactpro.th2.sim.rule.IRuleContext
 import com.exactpro.th2.sim.rule.impl.MessageCompareRule
 import java.time.LocalDateTime
 import java.util.Queue
-import java.util.LinkedList
+import java.util.concurrent.ConcurrentLinkedQueue
 import java.util.concurrent.atomic.AtomicInteger
 
 class KotlinFIXRule(field: Map<String, Any?>) : MessageCompareRule() {
     private val alias1 = "fix-demo-server1"
     private val alias2 = "fix-demo-server2"
     private val aliasdc1 = "dc-demo-server1"
     private val aliasdc2 = "dc-demo-server2"
+    private val lock = Any()
+    private val buyOrdersAndIds: Queue<OrderWithId> = ConcurrentLinkedQueue()
+    private val sellOrdersAndIds: Queue<OrderWithId> = ConcurrentLinkedQueue()
```

Comment:
Sorry, I don't understand logic 
You clear and remove elements under lock, but add without lock.
let's choose lock way in all cases, because the rule has two collection which should changed atomically.

Reply:
I've explained this above: "It's not necessary to use locking for adding new order to queue because it can't affect processing previously added orders (but we need to use concurrent version of queue). We only need locking to ensure three consequent orders from two queues will be extracted at once, if more orders will be added to queue tail at the same time this is not a problem. We also don't need to use locking with id counters as since using of AtomicInteger is enough to avoid duplicated ids."